src/dijkstra_spath.jl

```diff
@@ -96,6 +98,7 @@ end
 function set_source!{V,D}(state::DijkstraStates{V,D}, g::AbstractGraph{V}, s::V)
     i = vertex_index(s, g)
     state.parents[i] = s
+    state.hasparent[i] = true
```

Comment:
Could it be that this is the only line that needs changing to make sure hasparent on source = false? :)